kubernetes/resource_kubernetes_service_account.go

```diff
@@ -104,11 +105,15 @@ func resourceKubernetesServiceAccountCreate(d *schema.ResourceData, meta interfa
 	})
 
 	diff := diffObjectReferences(svcAcc.Secrets, resp.Secrets)
+	defaultSecret := diff[0]
 	if len(diff) > 1 {
-		return fmt.Errorf("Expected 1 generated default secret, %d found: %s", len(diff), diff)
+		for _, secret := range diff {
+			if strings.Contains(secret.Name, "token") {
```

Comment:
I believe this behavior should probably be traced in the logs at the `WARN` level and be documented on the resource page.

Reply:
Added